python/svix/api/common.py

```diff
@@ -44,3 +49,111 @@ class ApiBase:
 
     def __init__(self, client: AuthenticatedClient) -> None:
         self._client = client
+
+    def _get_httpx_kwargs(
+        self,
+        method: str,
+        path: str,
+        path_params: t.Optional[t.Dict[str, str]],
+        query_params: t.Optional[t.Dict[str, str]],
+        header_params: t.Optional[t.Dict[str, str]],
+        json_body: t.Optional[t.Dict[str, t.Any]],
+    ) -> t.Dict[str, t.Any]:
+        if path_params is None:
+            url = f"{self._client.base_url}{path}"
+        else:
+            formatted_path = path.format(**path_params)
+            url = url = f"{self._client.base_url}{formatted_path}"
```

Comment:
```suggestion
        if path_params is not None:
            path = path.format(**path_params)
        url = f"{self._client.base_url}{path}"
```
Essentially what you want is: if the path params are not none you need to pre process the path, otherwise you don't. Then you take the path and create a URL from it, so this is really what you want.